CommonMark/Formatters/HtmlFormatterSlim.cs

```diff
@@ -651,6 +659,33 @@ private static void InlinesToHtml(HtmlTextWriter writer, Inline inline, CommonMa
             }
         }
 
+        private static void WriteGitHubHeadingIds(HtmlTextWriter writer, Inline inline, CommonMarkSettings settings, Stack<InlineStackEntry> stack) {
+            using (var tempWriter = new System.IO.StringWriter()) {
+                var tempWrapper = new HtmlTextWriter(tempWriter);
+                var tempStringBuilder = new StringBuilder();
+                InlinesToPlainText(tempWrapper, inline, stack);
+                string plaintextContent = tempWriter.ToString();
+
+                // Normalize plaintext content according to GitHub ID rules
+                plaintextContent = new Regex(@"[^\w\-\ ]").Replace(plaintextContent, "").Replace(" ", "-");
```

Comment:
The regexp here is the best way I can think of to emulate the behaviour of my reference Github implementation, which itself uses a (Ruby) regexp matching [^\w-\ ] and replacing them with empty. What would you suggest as an alternative? That regexp works fine in .NET too.